TackleTabby/Assets/Scripts/BackgroundEnvironment/Water/MovingWater.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using UnityEngine;
+
+public class MovingWater : MonoBehaviour
+{
+    [SerializeField]
+    private RectTransform[] MovingPanels;
+    [SerializeField]
+    private float MovingSpeed = 25f;
+    [SerializeField]
+    private NonFlaggedDirections MovingDirection = NonFlaggedDirections.Right;
+    
+    private RectTransform RectTransform => (RectTransform)transform;
+    private float Width => RectTransform.rect.width;
+    private float Height => RectTransform.rect.height;
+
+    private float _progression = 0;
+
+    private void OnValidate()
+    {
+        UpdatePosition(0);
+        SetPanelWidth();
+    }
+
+    private void Update()
+    {
+        _progression = (_progression + CalculateAbsoluteProgressionDelta(MovingSpeed) * Time.deltaTime) % 1;
+        UpdatePosition(_progression * Width);
+        SetPanelWidth();
+    }
+
+    private float CalculateAbsoluteProgressionDelta(float absoluteSpeed)
+    {
+        return absoluteSpeed / Width; 
+    }
+
+    private void SetPanelWidth()
+    {
+        int panelCount = MovingPanels.Length;
+        for (int i = 0; i < panelCount; i++)
+        {
+            RectTransform rect = MovingPanels[i].transform as RectTransform;
+            rect.SetSizeWithCurrentAnchors(RectTransform.Axis.Horizontal, Width);
+            rect.SetSizeWithCurrentAnchors(RectTransform.Axis.Vertical, Height);
+        }
+    }
+
+    private void UpdatePosition(float offset)
+    {
+        switch (MovingDirection)
+        {
+            case NonFlaggedDirections.Right:
+                UpdateToRight(offset);
+                break;
+            case NonFlaggedDirections.Left:
+                UpdateToLeft(offset);
+                break;
```

Comment:
put an empty space between these two cases for the sake of consistency
```suggestion
            case NonFlaggedDirections.Right:
                UpdateToRight(offset);
                break;
    
            case NonFlaggedDirections.Left:
                UpdateToLeft(offset);
                break;
```